x/delayedack/keeper/finalize.go

```diff
@@ -55,6 +55,7 @@ func (k Keeper) finalizeRollappPacket(
 
 	switch rollappPacket.Type {
 	case commontypes.RollappPacket_ON_RECV:
+		// TODO: makes more sense to modify the packet when calling the handler, instead storing in db "wrong" packet
```

Comment:
@omritoptix 
I think we thought of that but I can't remember exactly why we didn't do it at the time
Maybe we were just in a rush I think
Do you remember?